MdeModulePkg/Include/Protocol/PciDoe.h

```diff
@@ -0,0 +1,73 @@
+/** @file
+  EFI PCI I/O Protocol provides the basic Memory, I/O, PCI configuration,
+  and DMA interfaces that a driver uses to access its PCI controller.
+
+  Copyright (c) 2006 - 2018, Intel Corporation. All rights reserved.<BR>
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#ifndef PCI_PROTOCOL_DOE_H_
+#define PCI_PROTOCOL_DOE_H_
+
+typedef struct _EFI_PCI_IO_DOE EFI_PCI_IO_DOE;
+
+/**
+  Check if a specific Vendor ID and Data Object Type is supported
+
+  @param  This                  A pointer to the EFI_PCI_IO_PROTOCOL instance.
+  @param  VendorId              The Vendor ID to check against
+  @param  Type                  The Data Object Type to check against
+
+  @retval EFI_SUCCESS           The operation on the PCI controller's attributes was completed.
+  @retval EFI_INVALID_PARAMETER One or more parameters are invalid.
+  @retval EFI_UNSUPPORTED       DOE is not supported or the VendorId and Type are not supported
+
+**/
+typedef
+EFI_STATUS
+(EFIAPI *EFI_PCI_IO_DOE_TYPE_CHECK)(
+  IN EFI_PCI_IO_PROTOCOL              *This,

```

Comment:
Please always use the protocol instance itself, here it should be EDKII_PCI_DOE_PROTOCOL.

The PCI bus can install EDKII_PCI_DOE_PROTOCOL and EFI_PCI_IO_PROTOCOL to same handle.
As such, the consumer can get the corresponding EDKII_PCI_DOE_PROTOCOL or EFI_PCI_IO_PROTOCOL.